src/vivarium_census_prl_synth_pop/utilities.py

```diff
@@ -548,3 +548,52 @@ def load_nicknames_data():
     nicknames = nicknames.apply(lambda x: x.astype(str).str.title()).set_index("name")
     nicknames = nicknames.replace("Nan", np.nan)
     return nicknames
+
+
+def extract_metadata_proportions(final_output_dir: Path) -> None:
+    # Get directories for each dataset which will contain all shard metadata files
+    dataset_metadata_dfs = []
+    directories = [
+        x.name for x in final_output_dir.iterdir() if x.is_dir() and x.name != "logs"
+    ]
+    for directory in directories:
+        dataset_directory = final_output_dir / directory
+        shard_metadata_files = sorted(list(chain(*[dataset_directory.glob("*.csv")])))
+        metadata_dfs = []
+        for shard_metadata_file in shard_metadata_files:
+            metadata_df = pd.read_csv(shard_metadata_file)
+            metadata_dfs.append(metadata_df)
+        dataset_metadata = pd.concat(metadata_dfs)
+        # Aggregate metadata
+        groupby_cols = ["dataset", "state", "year"]
+        aggregate_cols = [col for col in dataset_metadata.columns if col not in groupby_cols]
+        aggregated_metadata = (
```

Comment:
I think to answer your question about recording aggregated metadata, you will need to do a groupby three times, but the operation is the same as what you have below. Group by `["dataset", "state", "year"]` as you have here, but also do the same when grouping by `["dataset", "state"]` and `["dataset", "year"]`. In the aggregated case, set the excluded column equal to "all" or something similar. 

If you are concerned about the inherent duplication we'd get in the sample data since all rows will have the same state, you could the groupby to when we grouped by all the groupby columns, and not do "all" if they are identical, but I think it's actually correct to record the metadata for "US" and "all" separately even though the values will be identical, since the user could get there by having filtered to the "state" or by asking for "all states".